cucumber-expressions/ruby/lib/cucumber/cucumber_expressions/tree_regexp.rb

```diff
@@ -15,36 +15,36 @@ def initialize(regexp)
         non_capturing_maybe = false
         char_class = false
 
-        @regexp.source.split('').each_with_index do |c, n|
-          if c == '[' && !escaping
```

Comment:
Please don't make changes (such as renames) unless they are related to fixing the bug. Smaller diffs makes it much easier to review and port to other languages.

Reply:
Fixed this up, left the top bit as a slight optimisation